### ISSUE TITLE
Vaadin 15 does not redirect to error view if exception occurs during navigation

### ISSUE THREAD

**Issue by Bcricholf** [vaadin/flow#8430]
### Description of the bug

With Vaadin 15, JavaScriptBootstrapUI became the default UI. But when navigate, if an exception occurs during navigation (handleNavigation) she is not catch like Vaadin 14. (Router -> handleExceptionNavigation). After that, its impossible to use the navigation because a boolean are always set to true (navigationInProgress).
### Minimal reproducible example

Project with vaadin 14 where exception are correctly catch with rerouting to error view.
[my-starter-project-navigation-exception-handling-v14.zip](https://github.com/vaadin/flow/files/4688573/my-starter-project-navigation-exception-handling-v14.zip)

Project with vaadin 15 where exception are not catch and where navigation no longer works after this exception.
[my-starter-project-navigation-exception-handling-v15.zip](https://github.com/vaadin/flow/files/4688579/my-starter-project-navigation-exception-handling-v15.zip)

Both projects have the same code for view, only AppShellConfigurator are different.

### Expected behavior

If exception is throw during navigation, she would be catch and the navigation redirect to associate error view like in Vaadin 14.
### Versions:
    - Vaadin / Flow version (without bug): 14.1.25
    - Vaadin / Flow version (with bug): 15.0.6
    - Java version: 11.0.4
    - OS version: Windows 10
    - Browser version): Chrom Version 83.0.4103.6

**Comment by haijian-vaadin** [comment]
Hi @Bcricholf, thanks for the good ticket, we were able to reproduce the problem, will start working on it soon.

**Comment by haijian-vaadin** [comment]
Hi @Bcricholf, while we are fixing it, we found that it works if you use a RouterLink to do the navigation instead of a button click listener.
`RouterLink link = new RouterLink("AnotherView", AnotherView.class);`
instead of
```
Button button = new Button("Go to another view", e -> {
            UI.getCurrent().navigate(AnotherView.class);
        });
```
Would it be a temporary workaround for you?

**Comment by Bcricholf** [comment]
Hello @haijian-vaadin, I hadn't thought of this possibility. It could work for the majority of our cases, but we have a few use cases where we don't call navigate directly (With form in particular).
But for the moment we have no urgency to pass to Vaadin 15.
Thank you nevertheless for your proposal.

**Comment by tuscland** [comment]
Hello @haijian-vaadin,

Do you have an update about this?
We are planning to upgrade to Vaadin 17.

Best,
Camille

**Comment by haijian-vaadin** [comment]
Hi @tuscland & @Bcricholf, sorry for keeping you waiting, the issue was on hold for a while, I made a fix at #8866, you can expect a  new release soon.
